### PR TITLE
New thread densities

### PR DESCRIPTION
**Pull request by LunarX**
Depends on https://github.com/Infomaniak/android-core/pull/31

### CODE REVIEW

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 1]
```diff
@@ -129,6 +130,7 @@ class ThreadListAdapter(
     private fun Thread.getDisplayedRecipient(): Recipient = getDisplayedMessage().from.first()
 
     private fun CardviewThreadItemBinding.displayThread(thread: Thread) = with(thread) {
+        setupDensityDependentUi()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:133]:
```suggestion
        setupThreadDensityDependentUi()
```

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 2]
```diff
@@ -151,10 +153,14 @@ class ThreadListAdapter(
         if (unseenMessagesCount == 0) setThreadUiRead() else setThreadUiUnread()
 
         root.setOnClickListener { onThreadClicked?.invoke(this@with) }
+    }
+
+    private fun CardviewThreadItemBinding.setupDensityDependentUi() {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:158]:
```suggestion
    private fun CardviewThreadItemBinding.setupThreadDensityDependentUi() {
```

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 3]
```diff
@@ -173,17 +179,27 @@ class ThreadListAdapter(
...
-        newMailBullet.isGone = true
+        newMailBullet.isInvisible = true
     }
 
     private fun CardviewThreadItemBinding.setThreadUiUnread() {
         newMailBullet.isVisible = true
     }
 
     private fun ItemThreadDateSeparatorBinding.displayDateSeparator(title: String) {
+        setupDensityDependentUi()
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:190]:
```suggestion
        setupDateSeparatorDensityDependentUi()
```

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 4]
```diff
@@ -173,17 +179,27 @@ class ThreadListAdapter(
...
         sectionTitle.text = title
     }
 
+    private fun ItemThreadDateSeparatorBinding.setupDensityDependentUi() {
```
Comment (KevinBoulongne) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:194]:
```suggestion
    private fun ItemThreadDateSeparatorBinding.setupDateSeparatorDensityDependentUi() {
```

File: app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt [hunk 5]
```diff
@@ -301,6 +317,12 @@ class ThreadListAdapter(
         const val SWIPE_ANIMATION_THRESHOLD = 0.15f
         private val CARD_ELEVATION = 6.toPx().toFloat()
         private val CARD_CORNER_RADIUS = 12.toPx().toFloat()
+        private val THREAD_MARGIN_LARGE = 12.toPx()
+        private val THREAD_MARGIN_OTHER = 8.toPx()
+        private val DATE_TOP_MARGIN_LARGE = 4.toPx()
+        private val DATE_BOTTOM_MARGIN_LARGE = 4.toPx()
+        private val DATE_TOP_MARGIN_OTHER = 8.toPx()
```
Comment (JorisBodin) [inline app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:324]: Why don't you get them directly from the dimens?

Comment (LunarX) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:324]: We need access to a context to do so, and passing a context up to here is kinda unpractical. What do you think? Should we still do it?

Comment (sirambd) [reply app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListAdapter.kt:324]: Why would you need to pass a context ?, you have views available, so you can use their contexts

File: app/src/main/res/layout/cardview_thread_item.xml [hunk 6]
```diff
@@ -51,171 +54,164 @@
...
+                app:layout_constraintEnd_toStartOf="@id/iconCalendar"
+                app:layout_constraintStart_toEndOf="@id/mailSubject"
+                app:layout_constraintTop_toTopOf="@id/iconFavorite"
+                app:tint="@color/primaryTextColor"
+                tools:visibility="visible" />
+
+            <ImageView
+                android:id="@+id/iconFavorite"
+                android:layout_width="16dp"
+                android:layout_height="16dp"
```
Comment (sirambd) [inline app/src/main/res/layout/cardview_thread_item.xml:190]: Why you don't use `mediumIconSize`

**Review by sirambd — CHANGES_REQUESTED** [review]

**Pull request merged**